reminders.py

```diff
@@ -0,0 +1,350 @@
+#!/bin/python
+#
+# This script is used primarily to help progress Bitbucket Pull Requests that
+# have gone 'stale' and need attention.
+#
+# The Bitbucket 1.0 REST/BUILD API is used to firstly status Bitbucket, then
+# GET all OPEN Pull Requests, fetch details regarding their activity and POST
+# a relevant comment to alert authors/reviewers with suggestions to progress
+# the review. Build status against the Pull Request is also fetched if required.
+#
+
+import datetime
+import requests
+import json
+
+BITBUCKET_URL = 'http://localhost:80'
+PROJECT = 'PRJ'
+REPOSITORY = 'REPO'
+
+BUILD_ENDPOINT = '/rest/build-status/1.0/commits/'
+STATUS_ENDPOINT = '/rest/api/1.0/application-properties'
+REST_ENDPOINT = '/rest/api/1.0/projects/' + PROJECT + '/repos/' + REPOSITORY
+PULL_REQUESTS_ENDPOINT = '/pull-requests/'
+
+ACTIVITIES = '/activities'
+COMMENTS = '/comments'
+COMMITS = '/commits'
+PULL_REQUESTS = '/pull-requests'
+
+# Optional request headers
+REQUEST_GET_HEADER = ''
+REQUEST_POST_HEADER = ''
+
+# Datetime Constants
+TODAY = datetime.datetime.today()
+TWODAYS = datetime.timedelta(days=2)
+TENDAYS = datetime.timedelta(days=10)
+
+def main():
+    # Check Bitbucket is accessible
+    statusBitbucket();
+
+    # Fetch open PRs and extract ids
+    prIDs = fetchOpenPRs();
+
+    # Iterate over PR ID values
+    for prID in prIDs:
+        # Fetch content for each PR
+        url = BITBUCKET_URL + REST_ENDPOINT + PULL_REQUESTS_ENDPOINT + str(prID);
+        pullRequest = queryBitbucket(url);
+        pullRequestJSON = pullRequest.json();
+
+        # Calculate time since last update to PR
+        checksRequiredPR = analyzePRTimes(prID, pullRequestJSON);
+
+        # Calculate last time since comment/review action
+        checksRequiredAction = analyzeActionTimes(prID, url);
+
+        # Run checks
+        if (checksRequiredPR and checksRequiredAction):
+            if checkBuildStatus(prID, url):
+                continue;
+            elif checkReviewers(prID, pullRequestJSON):
+                continue;
+            elif checkApprovals(prID, pullRequestJSON):
+                continue;
+            elif checkMerge(prID, pullRequestJSON):
+                continue;
+            else:
+                # No action required
+                continue;
+
+    # Return success
+    exit(0);
+
+
+# Perform GET Request
+def queryBitbucket(url):
+    try:
+        if REQUEST_GET_HEADER:
+            response = requests.get(url, REQUEST_GET_HEADER);
+        else:
+            response = requests.get(url);
+    except:
+        exit(1);
+
+    # Exit with failure if bad status code returned
+    if (response.status_code != requests.codes.ok):
+        exit(1);
+
+    return response;
+
+
+# Perform POST
+def postToBitbucket(prID, comment):
+    # Prepare to POST
+    url = BITBUCKET_URL + REST_ENDPOINT + PULL_REQUESTS_ENDPOINT + str(prID) + COMMENTS;
+    data = { "text": comment };
```

Comment:
with the comment, what user does it appear as? some "bot" user? 
can we identify the Author and tag them via @ to send them a notification (that they'll probably ignore 😂)

Reply:
I'm hoping the way I have built the comment that it will be parsed by Bitbucket correctly (with the @ prefix)
The Jenkins user is what I've tested API calls against (hence the tasteful comment format :P)

Reply:
Ah I missed the nuanced posh-English-butler speak!